clients/src/main/java/org/apache/kafka/common/requests/MetadataRequest.java

```diff
@@ -79,10 +80,11 @@ public boolean isAllTopics() {
         }
 
         public List<String> topics() {
-            return data.topics()
-                .stream()
-                .map(MetadataRequestTopic::name)
-                .collect(Collectors.toList());
+            List<String> topicList = new ArrayList<>();
```

Comment:
nitt - if you know the exact size of the resulting collection - supply it tto the constructor

Reply:
fun thing: `MetadataRequestData` allows `topics` to be `null`. I will add a null check here and then, use the size. 
This method is used only in testing as of today.